tests/test_consumer.py

```
@@ -0,0 +1,312 @@
+                        messages.append(message)
+            except asyncio.TimeoutError:
+                pass
+
+            # Should handle empty stream gracefully
+            assert len(messages) == 0
+
+    @pytest.mark.asyncio
+    async def test_consumer_error_recovery(self, endpoint_url):
+        """Test consumer error recovery."""
+        # Test with non-existent stream
+        with pytest.raises(Exception):
```

Ruff (0.11.9)

Do not assert blind exception: `Exception`

(B017)